app/uk/gov/hmrc/trackingconsentfrontend/controllers/CookieSettingsController.scala

```diff
@@ -21,16 +21,40 @@ import play.api.mvc._
 import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
 import uk.gov.hmrc.trackingconsentfrontend.config.AppConfig
 import uk.gov.hmrc.trackingconsentfrontend.views.html.CookieSettingsPage
+import uk.gov.hmrc.trackingconsentfrontend.views.html.NotFoundPage
 
 import scala.concurrent.Future
 
 @Singleton
-class CookieSettingsController @Inject()(appConfig: AppConfig, mcc: MessagesControllerComponents, cookieSettingsPage: CookieSettingsPage)
+class CookieSettingsController @Inject()(
+  appConfig: AppConfig,
+  mcc: MessagesControllerComponents,
+  cookieSettingsPage: CookieSettingsPage,
+  notFoundPage: NotFoundPage)
     extends FrontendController(mcc) {
 
   implicit val config: AppConfig = appConfig
 
-  val cookieSettings: Action[AnyContent] = Action.async { implicit request =>
-    Future.successful(Ok(cookieSettingsPage()))
+  private val featureCookieName   = appConfig.featureCookieName
+  private val featureEnabledValue = appConfig.featureEnabledValue
+  private val featureToggleCookie = Cookie(name = featureCookieName, value = featureEnabledValue, httpOnly = false)
```

Comment:
I would move the creation of this cookie to closer to where it is needed (if needed), between current lines 51 and 52.